arkane/reference.py

```diff
@@ -0,0 +1,180 @@
+
+        Args:
+            species (rmgpy.molecule.Species): Molecule object representing the reference species
+            smiles (str): SMILES string representing the reference species
+            adjacency_list (str): An RMG adjacency list representation of the reference species
+            inchi (str): InChI string representing the reference species
+            reference_data (dict): Formatted as {'source_string': ReferenceDataEntry, ...}
+            calculated_data (dict): Formatted as {'model_chemistry': CalculatedDataEntry, ...}
+            preferred_reference (str): The source string key for the reference data to use for isodesmic reactions
+            index (int): Index of this species in the database of reference species located at
+                `RMG-database/input/reference_sets/`
+            label (str): A user defined label for easily identifying the species
+            cas_number (str): CAS number associated with the reference species
+            symmetry_number (int): The true symmetry number of the species (if not provided will default to the number
+                calculated by RMG)
+            **kwargs: Arguments passed to the parent ArkaneSpecies class when loading from a YAML file. Not intended for
+                user input
+        """
+
+        if species is None:
+            if smiles:
+                species = Species(smiles=smiles)
+            elif inchi:
+                species = Species(inchi=inchi)
+            elif adjacency_list:
+                species = Species().from_adjacency_list(adjacency_list)
+            else:
+                raise ValueError('Either an rmgpy species object, smiles string, InChI string, or an adjacency list '
+                                 'must be given to create a ReferenceSpecies object')
+
+        super(ReferenceSpecies, self).__init__(species=species, label=label, **kwargs)
+
+        self.reference_data = reference_data
+        self.calculated_data = calculated_data
+        self.index = index
+        self.cas_number = cas_number
+        self.preferred_reference = preferred_reference
+
+        # Alter the symmetry number calculated by RMG to the one provided by the user
+        if symmetry_number:
+            self.symmetry_number = symmetry_number
+
+    def __repr__(self):
+        if self.index:
+            label = '{0}({1})'.format(self.smiles, self.index)
+        else:
+            label = '{0}'.format(self.smiles)
+
+        return '<ReferenceSpecies {0}>'.format(label)
+
+    @property
+    def reference_data(self):
+        return self._reference_data
+
+    @reference_data.setter
+    def reference_data(self, value):
+        if not value:
+            self._reference_data = {}
+        elif isinstance(value, dict):
+            if all(isinstance(source, str) for source in value.keys()):
+                if all(isinstance(data_entry, ReferenceDataEntry) for data_entry in value.values()):
```

Comment:
If these if-statements are not true, `self._reference_data` does not get set and no error is raised. Is that desired?